src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -271,14 +271,19 @@ protected override ImmutableArray<PendingBranch> RemoveReturns()
 
         protected virtual void ReportUnassignedOutParameter(ParameterSymbol parameter, SyntaxNode node, Location location)
         {
-            if (!_requireOutParamsAssigned && topLevelMethod == currentMethodOrLambda) return;
+            if (!_requireOutParamsAssigned && topLevelMethod == currentMethodOrLambda ||
+                node == null && location == null)
```

Comment:
Unclear why we need to guard against `null` in these two cases.  Can you elaborate?